sensor/tests/resource/helper.go

```diff
@@ -287,3 +316,37 @@ func (c *TestContext) waitForResource(timeout time.Duration, fn condition) error
 		}
 	}
 }
+
+func GetLastMessageWithDeploymentName(messages []*central.MsgFromSensor, n string) *central.MsgFromSensor {
+	var lastMessage *central.MsgFromSensor
+	for i := len(messages) - 1; i > 0; i-- {
+		if messages[i].GetEvent().GetDeployment().GetName() == n {
+			lastMessage = messages[i]
+			break
+		}
+	}
+	return lastMessage
+}
+
+func GetUniquePodNamesFromPrefix(messages []*central.MsgFromSensor, prefix string) []string {
+	uniqueNames := set.NewStringSet()
+	for _, msg := range messages {
+		if msg.GetEvent().GetPod() != nil {
+			pod := msg.GetEvent().GetPod()
+			if strings.Contains(pod.GetName(), prefix) {
+				uniqueNames.Add(pod.GetName())
+			}
+		}
+	}
+	return uniqueNames.AsSlice()
+}
+
+func GetUniqueDeploymentNames(messages []*central.MsgFromSensor) []string {
+	uniqueNames := set.NewStringSet()
+	for _, msg := range messages {
+		if msg.GetEvent().GetDeployment() != nil {
```

Comment:
Same here. I think we should check namespace.

Reply:
Done